### PR TITLE
Move `order_by` from query to select

### PR DESCRIPTION
**Pull request by devgony**
To implement #417 , #418 
We decided to move `order_by` from `Query` to `Select` first. 

At sqlparser::ast::query, `order_by` is on `Query`,
but from now, our translated gluesql::ast::query will consider `order_by` as on `Select`.

### CODE REVIEW

File: src/plan.rs [hunk 1]
```diff
@@ -83,13 +80,13 @@ async fn plan_query<T: Debug>(storage: &dyn Store<T>, query: Query) -> Result<Qu
...
                 body: SetExpr::Select(select),
-                order_by,
                 limit,
                 offset,
             });
         }
     };
 
+    let Select { order_by, .. } = select.as_ref();
     let index = order_by.last().and_then(|value_expr| {
```
Comment (panarch) [inline src/plan.rs:90]: you can simply use `select.order_by` without destructuring
```rust
let index = select.order_by.last().and_then(|value_expr| {
 ..
```

File: src/plan.rs [hunk 2]
```diff
@@ -124,11 +122,11 @@ async fn plan_query<T: Debug>(storage: &dyn Store<T>, query: Query) -> Result<Qu
                 selection,
                 group_by,
                 having,
+                order_by,
             };
 
             Ok(Query {
                 body: SetExpr::Select(Box::new(select)),
-                order_by: Vector::from(order_by).pop().0.into(),
```
Comment (panarch) [inline src/plan.rs:131]: Here is the key of `index_order_by` failure.
If the planner succeeded to find the index in `ORDER BY`, then it needs to remove the last order_by item which is consumed by the index.

Comment (devgony) [reply src/plan.rs:131]: Thanks.. I realized we should never remove previous logic while refactoring...

**Review by panarch — APPROVED** [review]
Looks all good!! thanks a lot, merging 👍

**Pull request merged**